docs/book/v2/routing-middleware.md

```diff
@@ -0,0 +1,265 @@
+# Routing Middleware
+
+Apart from PSR-15 `Psr\Http\Server\MiddlewareInterface` or `Psr\Http\Server\RequestHandlerInterface` middleware or
+service name strings which resolve to such instances, the `middleware` key in the route definition takes also an
+instance of the `PipeSpec` class.
+
+Here are some examples how middleware can be dispatched:
+
+## `PipeSpec` with PSR-15 Service Names
+
+Middleware can be passed as a `PipeSpec` object with container service class-string literals. This is the preferred
+method.
+
+Assuming you have two middlewares, `Application\Middleware\SelectLanguageMiddleware` which determines the language of
+the visitor and `Application\Middleware\AlbumFromRouteMiddleware` which fetches an `Album` object by the ID in the route
+param. Both of them add this information to the `ServerRequestInterface` attributes which can be used by the
+`Application\Handler\AlbumDetailHandler` at the end of the pipe:
+
+```php
+use Application\Handler\AlbumDetailHandler;
+use Application\Middleware\AlbumFromRouteMiddlware;
+use Application\Middleware\SelectLanguageMiddleware;
+use Laminas\Mvc\Middleware\PipeSpec;
+use Laminas\Router\Http\Segment;
+
+return [
+    'router' => [
+        'routes' => [
+            'album-details' => [
+                'type' => Segment::class,
+                'options' => [
+                    'route' => '/album/:album_id',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => new PipeSpec(
+                            SelectLanguageMiddleware::class,
+                            AlbumFromRouteMiddelware::class,
+                            AlbumDetailHandler::class
+                        ),
+                    ],
+                ],
+            ],
+        ],
+    ],
+];
+```
+
+## `MiddlewareInterface` or `RequestHandlerInterface` Service Strings
+
+Alternatively, if you only dispatch one class, you can pass a class-string or alias which resolves to a
+`MiddlewareInterface`, `RequestHandlerInterface` or `Laminas\Stratigility\MiddlewarePipe`.
```

Comment:
I find this confusing. Alternatively to what? I would prefer this to be a bit more explicit:

```suggestion
If you only dispatch one class, you can pass a container service class-string or alias literal which resolves to a PSR-15 `Psr\Http\Server\MiddlewareInterface` or `Psr\Http\Server\RequestHandlerInterface` middleware.

`Laminas\Stratigility\MiddlewarePipe` implements both interfaces and could be provided by the application container as a configured instance. It is what is used internally for the `PipeSpec` definitions.
```